python/adjoint/optimization_problem.py

```diff
@@ -433,10 +462,22 @@ def _stop(sim):
                         current_fields[mi][ic] = mp.get_fluxes(m)[fcen_idx]
                     elif isinstance(m,mp.DftFields):
                         current_fields[mi][ic] = atleast_3d(sim.get_dft_array(m, cc, fcen_idx))
+
+                    elif isinstance(m, mp.simulation.DftNear2Far):
+                        current_fields[mi][ic] = atleast_3d(sim.get_dft_array(m, cc, fcen_idx))
                     else:
                         raise TypeError("Monitor of type {} not supported".format(type(m)))
-                    relative_change_raw = np.abs(previous_fields[mi][ic] - current_fields[mi][ic]) / np.abs(previous_fields[mi][ic])
-                    relative_change.append(np.mean(relative_change_raw.flatten())) # average across space and frequency
+
+                    #relative_change_raw = np.abs(previous_fields[mi][ic] - current_fields[mi][ic]) / np.abs(previous_fields[mi][ic])
+                    #relative_change.append(np.mean(relative_change_raw.flatten())) # average across space and frequency
+
+                    if previous_fields[mi][ic] is not None:
+                        relative_change_raw = np.abs(previous_fields[mi][ic] - current_fields[mi][ic]) / np.abs(previous_fields[mi][ic])
+                        relative_change.append(np.mean(relative_change_raw.flatten())) # average across space and frequency
+                    else:
+                        relative_change.append(1)
+
+
```

Comment:
This modification doesn't look right... what's the advantage of doing it this way?